lib/std/comptime_hash_map.zig

```diff
@@ -0,0 +1,129 @@
+const std = @import("std.zig");
+const hash_map = std.hash_map;
+const testing = std.testing;
+const math = std.math;
+
+/// A comptime hashmap constructed with automatically constructed hash and eql functions.
+pub fn AutoComptimeHashMap(comptime K: type, comptime V: type, comptime values: var) type {
+    return HashMap(K, V, hash_map.getAutoHashFn(K), hash_map.getAutoEqlFn(K), values);
+}
+
+/// Builtin hashmap for strings as keys.
+pub fn ComptimeStringHashMap(comptime V: type, comptime values: var) type {
+    return ComptimeHashMap([]const u8, V, hash_map.hashString, hash_map.eqlString, values);
+}
+
+/// A hashmap which is constructed at compile time from constant values.
+/// Intended to be used as a faster lookup table.
+pub fn ComptimeHashMap(comptime K: type, comptime V: type, comptime hash: fn (key: K) u32, comptime eql: fn (a: K, b: K) bool, comptime values: var) type {
+    std.debug.assert(values.len != 0);
+
+    // TODO should this be configureable
+    @setEvalBranchQuota(1000 * values.len);
```

Comment:
This is good, and it is already configurable at the callsites of the functions, because `@setEvalBranchQuota` will never make the quota smaller, only bigger. So if this convenient heuristic is not correct, callsites can put their own calls to `@setEvalBranchQuota` to make it work. I think you can simply remove the TODO comment.